src/integrations/blocks/block-categories.php

```diff
@@ -20,7 +37,15 @@ public static function get_conditionals() {
 	 * {@inheritDoc}
 	 */
 	public function register_hooks() {
-		\add_filter( 'block_categories', [ $this, 'add_block_categories' ] );
+		$wordpress_version = $this->wordpress_helper->get_wordpress_version();
+
+		// The 'block_categories' filter has been deprecated in WordPress 5.8 and replaced by 'block_categories_all'.
+		if ( \version_compare( $wordpress_version, '5.8', '<' ) ) {
```

Comment:
I think we can always compare against `5.8-beta0`, since `5.8` is later than `5.8-beta0`. It would mean that it will also work for people using the beta (plus QA would not have to change the code themselves).
```suggestion
		if ( \version_compare( $wordpress_version, '5.8-beta0', '<' ) ) {
```
It would mean changing this for all four PR's though.

Reply:
Nice suggestion! Will implement this for all PR's and update the test instructions.